### PR TITLE
Appwebsocket authentication token from launcher env

### PR DESCRIPTION
**Pull request by matthme**
* takes the app websocket token from the launcher env if available
* removes deprecated framework specific zome call signing logic

### CODE REVIEW

**Comment by matthme** [comment]
converted it to a draft because I haven't actually tested it in practice yet

File: src/api/app/websocket.ts [hunk 1]
```diff
@@ -161,10 +159,7 @@ export class AppWebsocket implements AppClient {
    * @param options - {@link (WebsocketConnectionOptions:interface)}
    * @returns A new instance of an AppWebsocket.
    */
-  static async connect(
-    token: AppAuthenticationToken,
-    options: WebsocketConnectionOptions = {}
-  ) {
+  static async connect(options: AppWebsocketConnectionOptions = {}) {
```
Comment (ThetaSinner) [inline src/api/app/websocket.ts:162]: Changelog entry for this please

File: src/api/app/websocket.ts [hunk 2]
```diff
@@ -180,7 +175,17 @@ export class AppWebsocket implements AppClient {
     }
 
     const client = await WsClient.connect(options.url, options.wsClientOptions);
-    await client.authenticate({ token });
+    if (env?.APP_INTERFACE_TOKEN) {
+      await client.authenticate({ token: env.APP_INTERFACE_TOKEN });
```
Comment (ThetaSinner) [inline src/api/app/websocket.ts:181]: Maybe a comment here to note that the launcher must provide a `single_use = false` token?

File: src/api/app/websocket.ts [hunk 3]
```diff
@@ -426,14 +431,7 @@ const callZomeTransform: Transformer<
...
     } else {
-      const env = getLauncherEnvironment();
-      if (!env) {
-        return signZomeCall(request);
-      }
-      if (env.FRAMEWORK === "electron") {
-        return signZomeCallElectron(request);
-      }
-      return signZomeCallTauri(request);
+      return signZomeCall(request);
```
Comment (ThetaSinner) [inline src/api/app/websocket.ts:436]: Why is zome call singing by the runtime being removed?

File: src/api/app/websocket.ts (hunk 1 above)
Comment (ThetaSinner) [reply src/api/app/websocket.ts:162]: Ahh, done while I was reviewing

File: src/api/app/websocket.ts (hunk 3 above)
Comment (matthme) [reply src/api/app/websocket.ts:436]: Only the legacy methods are removed. The framework agnostic `hostSigner.signZomeCall()` is still there.

File: src/api/app/websocket.ts (hunk 2 above)
Comment (matthme) [reply src/api/app/websocket.ts:181]: Hmm...good point. Do you see any problems with launcher providing `single_use = false` tokens?

File: src/api/app/websocket.ts (hunk 3 above)
Comment (ThetaSinner) [reply src/api/app/websocket.ts:436]: Ahh I see what you're saying, nice okay!

**Review by ThetaSinner — APPROVED** [review]

File: src/api/app/websocket.ts (hunk 2 above)
Comment (ThetaSinner) [reply src/api/app/websocket.ts:181]: The alternative would be providing a host call to retrieve a token each time an appws is created. I think `single_use = false` is fine to start and maybe we can consider retrieving tokens on demand as an enhancement later?

**Pull request merged**